src/Internal/RestApiUtil.php

```diff
@@ -0,0 +1,195 @@
+	 *
+	 * [
+	 *   "reason" => null,     (if it's missing or any empty value, set as null)
+	 *   "api_refund" => true, (if it's missing or non-bool, set as "true")
+	 *   "line_items" => [     (convert sequential array to associative based on "id")
+	 *     "111" => [
+	 *       "qty" => 222,     (rename "quantity" to "qty")
+	 *       "refund_total" => 333,
+	 *       "refund_tax" => [ (convert sequential array to associative based on "id" and "refund_total)
+	 *         "444" => 555,...
+	 *       ],...
+	 *   ]
+	 * ]
+	 *
+	 * It also calculates the amount if missing and whenever possible, see maybe_calculate_refund_amount_from_line_items.
+	 *
+	 * The conversion is done in a way that if the request is already in the internal format,
+	 * then nothing is changed for compatibility. For example, if the line items array
+	 * is already an associative array or any of its elements
+	 * is missing the "id" key, then the entire array is left unchanged.
+	 * Same for the "refund_tax" array inside each line item.
+	 *
+	 * @param \WP_REST_Request $request The request to adjust.
+	 */
+	public static function adjust_create_refund_request_parameters( \WP_REST_Request &$request ) {
+		if ( empty( $request['reason'] ) ) {
+			$request['reason'] = null;
+		}
+
+		if ( ! is_bool( $request['api_refund'] ) ) {
+			$request['api_refund'] = true;
+		}
+
+		if ( empty( $request['line_items'] ) ) {
+			$request['line_items'] = array();
+		} else {
+			$request['line_items'] = self::adjust_line_items_for_create_refund_request( $request['line_items'] );
+		}
+
+		self::maybe_calculate_refund_amount_from_line_items( $request );
+	}
+
+	/**
+	 * Calculate and set the "amount" parameter in the request based on the amounts found in line items.
+	 * This will ONLY happen if ALL of the following is true:
+	 *
+	 * - The request doesn't have an "amount" parameter already.
+	 * - "line_items" in the request is a non-empty array.
+	 * - All line items have a "refund_total" field with a numeric value.
+	 * - All values inside "refund_tax" in all line items are a numeric value.
+	 *
+	 * The request is assumed to be in internal format already.
+	 *
+	 * @param \WP_REST_Request $request The request to maybe calculate the total amount for.
+	 */
+	private static function maybe_calculate_refund_amount_from_line_items( &$request ) {
+		if ( isset( $request['amount'] ) ) {
+			return;
+		}
+
+		$line_items = $request['line_items'];
+
+		if ( ! is_array( $line_items ) || empty( $line_items ) ) {
+			return;
+		}
+
+		$amount = 0;
+
+		foreach ( $line_items as $item ) {
+			if ( ! isset( $item['refund_total'] ) || ! is_numeric( $item['refund_total'] ) ) {
+				return;
+			}
+
+			$amount += $item['refund_total'];
+
+			if ( ! isset( $item['refund_tax'] ) ) {
+				continue;
+			}
+
+			foreach ( $item['refund_tax'] as $tax ) {
+				if ( ! is_numeric( $tax ) ) {
+					return;
+				}
+				$amount += $tax;
+			}
+		}
+
+		$request['amount'] = strval( $amount );
```

Comment:
instead of setting it here, maybe we should return and set it in callee function? that way it will be more usable, wdyt?

Reply:
I think that the problem is with the method name, instead of `calculate` it should say `adjust`; but just returning the value and calling the method only if there's no `amount` present is also a valid option.